src/python/pants/option/global_options.py

```diff
@@ -114,6 +114,9 @@ def register_bootstrap_options(cls, register):
     # after -l and -q in help output, which is conveniently contextual.
     register('--colors', type=bool, default=sys.stdout.isatty(), recursive=True, daemon=False,
              help='Set whether log messages are displayed in color.')
+    register('--warning-filters', type=dict, default={},
+             help='Dict mapping action -> message, applying the `action` to warnings matching '
```

Comment:
I think we want to invert this mapping. The reason is that dictionaries cannot have duplicate keys, so you could not have multiple `ignore`s. 

Also, an example here would be extremely helpful, e.g. `--warnings-filters={"DeprecationWarning: this option has been removed": "ignore"}`.

Reply:
Hm, reading your code, it seems you expect messages to be a list? If so, rename `message` to `messages` and change the example to:

```
--warnings-filters={"ignore": ["DeprecationWarning: this option has been removed", "DeprecationWarning: pex no longer"]}
```

Reply:
This just seems too general to me. What about only supporting one implicit action - `ignore` - and just taking a list of of message regex strings; ie: `--ignore-pants-warnings="['...', '...']"`.

Reply:
Agreed. I prefer both this name and making the option a list instead of dict.

Reply:
Done!